rmgpy/rmg/model.py

```diff
@@ -431,12 +431,23 @@ def make_new_reaction(self, forward, check_existing=True, generate_thermo=True):
         """
 
         # Determine the proper species objects for all reactants and products
-        try:
+        if forward.family and forward.is_forward:
             reactants = [self.make_new_species(reactant, generate_thermo=generate_thermo)[0] for reactant in forward.reactants]
-            products = [self.make_new_species(product, generate_thermo=generate_thermo)[0] for product in forward.products]
-        except:
-            logging.error(f"Error when making species in reaction {forward:s} from {forward.family:s}")
-            raise
+            products = []
+            for product in forward.products:
+                spcs = self.make_new_species(product, generate_thermo=generate_thermo,check_decay=True)
+                if type(spcs) == tuple:
+                    products.append(spcs[0])
+                elif type(spcs) == list:
+                    products.extend([spc[0] for spc in spcs])
```

Comment:
I guess in some rare cases, there can be four or even five products in a reaction? Will that cause any problem elsewhere?